api/v1alpha1/kongplugin_installation.go

```diff
@@ -0,0 +1,129 @@
+
+// KongPluginInstallation allows to use a custom Kong Plugin distributed as a container image available in a registry.
+// Such plugin can be associated with GatewayConfiguration or DataPlane to be available for particular Kong Gateway
+// and to be configured with KongPlugin CRD.
+type KongPluginInstallation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KongPluginInstallationSpec   `json:"spec,omitempty"`
+	Status KongPluginInstallationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// KongPluginInstallationList contains a list of KongPluginInstallation.
+type KongPluginInstallationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KongPluginInstallation `json:"items"`
+}
+
+// KongPluginInstallationSpec defines the desired state of KongPluginInstallation.
+type KongPluginInstallationSpec struct {
+
+	// Image is OCI image URL for a packaged Custom Kong Plugin.
+	//
+	//+kubebuilder:validation:Required
+	Image string `json:"image"`
+
+	// SecretRef allows specifying secret to be used for OCI registry authentication to pull the image with custom Kong Plugin.
+	// The Secret format should follow https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry.
+	// When field is omitted it is assumed that the image is public and can be fetched without providing any credentials.
+	//
+	//+optional
+	SecretRef *corev1.SecretReference `json:"secretRef,omitempty"`
+}
+
+// KongPluginInstallationStatus defines the observed state of KongPluginInstallation.
+type KongPluginInstallationStatus struct {
+	// Conditions describe the current conditions of this KongPluginInstallation.
+	//
+	//+listType=map
+	//+listMapKey=type
+	//+kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// KongPluginInstallationConditionType is the type for status conditions on
+// KongPluginInstallation resources. This type should be used with the
+// KongPluginInstallationStatus.Conditions field.
+type KongPluginInstallationConditionType string
+
+// KongPluginInstallationConditionReason defines the set of reasons that explain why
+// a particular KongPluginInstallation condition type has been raised.
+type KongPluginInstallationConditionReason string
+
+const (
+	// This condition indicates whether the controller has fetched and make
+	// available for use a specific Custom Kong Plugin.
+	//
+	// It is a positive-polarity summary condition, and so should always be
+	// present on the resource with ObservedGeneration set.
+	//
+	// It should be set to Unknown if the controller performs updates to the
+	// status before it has all the information it needs to be able to determine
+	// if the condition is true.
+	//
+	// Possible reasons for this condition to be true are:
```

Comment:
```suggestion
	// Possible reasons for this condition to be True are:
```

Just to be consistent with other comments below.